js/main.js

```diff
@@ -80,19 +114,47 @@ function switchViews(view) {
       data.view = $viewElements[i].getAttribute('data-view');
     }
   }
+  if ($ul.childElementCount === 0) {
+    $noItems.className = 'center-text';
+  } else {
+    $noItems.className = 'center-text hidden';
+  }
+
 }
 
 window.addEventListener('DOMContentLoaded', appendDOM);
 var $noItems = document.querySelector('.center-text');
-if ($ul.childElementCount === 0) {
-  $noItems.className = 'center-text hidden';
-} else {
-  $noItems.className = 'center-text';
-}
+
 
 function appendDOM() {
   for (var i = 0; i < data.entries.length; i++) {
     entryDOM(data.entries[i]);
   }
   switchViews(data.view);
+  var $icon = document.querySelectorAll('i');
+  for(var i = 0; i < $icon.length; i++){
+    $icon[i].setAttribute("EntryId", $icon.length - i)
+    $icon[i].addEventListener('click', function(){
+      switchViews(event.target.getAttribute('data-view'))
+      console.log("$icon[i]:", $icon[0].attributes.entryid)
+    })
+  }
+  var $liAll = document.querySelectorAll('li');
+  for (var y = 0; y < $liAll.length; y++) {
+    $liAll[y].setAttribute('position', $liAll.length - y)
+  }
 }
+
+$entriesPage.addEventListener('click', function () {
+  if (event.target.tagName === 'I') {
+    for (var x = 0; x < data.entries.length; x++){
+      if (data.entries[x].entryId === parseInt(event.target.attributes.EntryId.value)){
+        $inputTitle.value = data.entries[x].title;
+        $imgUrl.value = data.entries[x].photoUrl
+        $textArea.value = data.entries[x].notes
+        $setImg.setAttribute('src', data.entries[x].photoUrl);
+        data.editing = parseInt(event.target.attributes.EntryId.value)
+      }
+    }
+  }
+})
```

Comment:
Why do you need the other event handler functions for `i` elements if you have this? You should be doing all the necessary work for setting up an entry to be edited inside of this event handler function.